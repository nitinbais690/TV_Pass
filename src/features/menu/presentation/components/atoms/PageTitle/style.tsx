import { scale } from 'qp-common-ui';
import { appPaddingValues } from 'core/styles/AppStyles';
import { StyleSheet } from 'react-native';

export const PageTitleStyle = (appColors: any) => {
    return StyleSheet.create({
        textContainer: {
            marginStart: appPaddingValues.lg,
            marginEnd: appPaddingValues.lg,
            marginTop: scale(47),
            flex: 1,
        },
        dotText: {
            color: appColors.brandTintLight,
        },
        textStyle: {
            color: appColors.pageTitleColor,
        },
    });
};
