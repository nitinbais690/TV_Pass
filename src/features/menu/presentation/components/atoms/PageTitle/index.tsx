import { appFontStyle } from 'core/styles/AppStyles';
import React from 'react';
import { Text, View } from 'react-native';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { PageTitleStyle } from './style';

/*
   Renders Page Title for Menu screens
 */
export default function PageTitle(props: { title: string }) {
    const style = PageTitleStyle(useAppColors());
    return (
        <View style={style.textContainer}>
            <Text style={[appFontStyle.header1, style.textStyle]}>
                {props.title}
                {props.title !== '' && <Text style={[appFontStyle.header1, style.dotText]}>.</Text>}
            </Text>
        </View>
    );
}

interface PageTitleprops {
    title: string;
}
