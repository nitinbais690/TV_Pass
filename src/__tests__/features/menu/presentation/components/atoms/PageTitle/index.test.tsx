import PageTitle from 'features/menu/presentation/components/atoms/PageTitle';
import React from 'react';
import { render, toJSON } from '@testing-library/react-native';

jest.mock('core/presentation/hooks/use-app-colors');

describe('Page Title.test', () => {
    it('Page title rendered correctly with style', () => {
        const { container } = render(<PageTitle title={'Menu'} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
