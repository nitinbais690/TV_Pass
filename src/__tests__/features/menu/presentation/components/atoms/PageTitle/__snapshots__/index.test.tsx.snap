// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Title.test Page title rendered correctly with style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginEnd": 28,
        "marginStart": 28,
        "marginTop": 68.48571428571428,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Black",
            "fontSize": 44,
            "fontWeight": "900",
          },
          Object {
            "color": "#ECECEC",
          },
        ]
      }
    >
      Menu
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ProximaNova-Black",
              "fontSize": 44,
              "fontWeight": "900",
            },
            Object {
              "color": "#FF6D2E",
            },
          ]
        }
      >
        .
      </Text>
    </Text>
  </View>
</View>
`;
